Assets/ModScripts/Submodules/MeteredButton.cs

```diff
@@ -0,0 +1,329 @@
+        float heldTime = Mathf.RoundToInt(Math.Abs(pressTime - releaseTime));
+
+        string actionChar = finalActions[stage][0].ToString();
+        int actionNum = Int32.Parse(finalActions[stage][1].ToString());
+        double lastDigit = Math.Floor(Module.Bomb.GetTime()) % 10;
+        if (actionChar == "T")
+        {
+            if (heldTime == 0f && lastDigit == actionNum)
+            {
+                Debug.LogFormat("[The Cruel Modkit #{0}] Button tapped correctly. Stage {1} passed.", ModuleID, stage+2);
+                Module.StartCoroutine(AdvanceStage());
+            }
+            else
+            {
+                Module.CauseStrike();
+                Debug.LogFormat("[The Cruel Modkit #{0}] Strike! Button tapped incorrectly at last digit {1} or button was held.", ModuleID, lastDigit);
+                Module.StartCoroutine(Strike());
+            }
+        }
+        else if (actionChar == "H") 
+        {
+            double holdDigit = Math.Floor(pressTime % 10);
+            double releaseDigit = Math.Floor(releaseTime % 10);
+            if (holdDigit == (actionNum + Info.NumberDisplay) % 10 && releaseDigit == Math.Abs(actionNum - Info.NumberDisplay))
+            {
+                Debug.LogFormat("[The Cruel Modkit #{0}] Button held correctly. Stage {1} passed.", ModuleID, stage + 1);
+                Module.StartCoroutine(AdvanceStage());
+            }
+            else
+            {
+                Module.CauseStrike();
+                Debug.LogFormat("[The Cruel Modkit #{0}] Strike! Button held incorrectly (held on {1}, released on {2}).", ModuleID, holdDigit, releaseDigit);
+                Module.StartCoroutine(Strike());
+            }
+        }
+        else if (actionChar == "M")
+        {
+            pressedNum++;
+            if (!mashing) 
+                Module.StartCoroutine(MashCount());
+        }
+        else if (actionChar == "X")
+        {
+            if (utilPresses == Info.NumberDisplay * actionNum)
+            {
+                Debug.LogFormat("[The Cruel Modkit #{0}] ❖ button pressed correctly. Stage {1} passed.", ModuleID, stage + 1);
+                Module.StartCoroutine(AdvanceStage());
+            }
+            else
+            {
+                Module.CauseStrike();
+                Debug.LogFormat("[The Cruel Modkit #{0}] Strike! ❖ button pressed incorrect number of times ({1} times).", ModuleID, utilPresses);
+                Module.StartCoroutine(Strike());
+            }
+        }
+        utilPresses = 0;
+    }
+
+    public override void OnUtilityPress()
+    {
+        if (Module.IsAnimating() || animating)
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.UtilityButton.GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.5f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
```

Comment:
This is check every time big diamond is pressed. Maybe a getter for the Solving variable could be created?